.github/workflows/npm.yml

```diff
@@ -0,0 +1,31 @@
+# This workflow runs a test build for npm against changes on main or PRs
+
+name: NPM Build
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: macos-latest
```

Comment:
Do we need to run this on mac? Why not a linux container? It would be faster I bet.